media/open_zim.ksy

```diff
@@ -0,0 +1,409 @@
+          first_blob_ptr_u_v:
+            value: |
+              (
+                info.compression == compression::lzma2
+                ?
+                first_blob_ptr_compressed_lzma
+                :
+                (
+                  info.compression == compression::zstd
+                  ?
+                  first_blob_ptr_compressed_zstd
+                  :
+                  first_blob_ptr_uncompressed
+                )
+              ).as<blob_ptr_u>.value
+          blob_count_plus_one:
+            value: first_blob_ptr_u_v / info.ptr_size
+          blob_count:
+            value: blob_count_plus_one - 1
+          blobs_ptrs:
+            value: |
+              (
+                info.compression == compression::lzma2
+              ?
+                blobs_ptrs_compressed_lzma2
+              :
+                (
+                  info.compression == compression::zstd
+                ?
+                  blobs_ptrs_compressed_zstd
+                :
+                  blobs_ptrs_uncompressed
+                )
+              ).ptrs
+            if: |
+              info.compression == compression::lzma2
+              or info.compression == compression::zstd
+              or info.compression == compression::none
+              or info.compression == compression::default
+        types:
+          blobs_ptrs:
+            params:
+              - id: count
+                type: u8
+            seq:
+              - id: ptrs
+                type: blob_ptr_u
+                repeat: eos
+            instances:
+              info:
+                value: _parent.info
+          blob:
+            params:
+              - id: idx
+                type: u8
+            instances:
+              data:
+                pos: _parent.ptr + _parent.blobs_ptrs[idx].as<blob_ptr_u>.value
+                size: _parent.blobs_ptrs[idx + 1].as<blob_ptr_u>.value - _parent.blobs_ptrs[idx].as<blob_ptr_u>.value
+          info_t:
+            seq:
+              - id: reserved
+                type: b3
+                valid: 0
+              - id: is_ptr_u8
+                -orig-id: extended
+                type: b1
+              - id: compression
+                type: b4
+                enum: compression
+                valid:
+                  expr: _ != compression::lzma2 and _ != compression::zstd and _ != compression::bzip2 and _ != compression::zlib
+                doc: LZMA and ZStd are not supported because of behavior of lack of support of streaming processors in KS. bzip2 and zlib are not supported because they are not supported in the official lib (though in principle they probably can be supported).
+            instances:
+              ptr_size:
+                value: "is_ptr_u8 ? 8 : 4"
+          blob_ptr_u:
+            seq:
+              - id: value
+                type:
+                  -affected-by: 561
```

Comment:
When I see the [diff showing changes](https://github.com/kaitai-io/kaitai_struct_formats/compare/32d136bbb99d2a9be4c28632c2cec0bc3a764e4d..dc857b989220be64efcd9e4e853a4dbe0f13f6d8) done in the last commit https://github.com/kaitai-io/kaitai_struct_formats/commit/dc857b989220be64efcd9e4e853a4dbe0f13f6d8:

```diff
-                   -affected-by: don't know how to call method 'info' of object type 'CalcKaitaiStructType'
+                   -affected-by: 561
```

how is https://github.com/kaitai-io/kaitai_struct/issues/561 related to the error message that you quoted there earlier?

Reply:
Seems to be a manifestation of the same issue - defective or unimplemented machinery for properly resolving types relative to `_parent`.